### PR TITLE
[4.1] Disable auth binding code introduced with 4.1.1

### PR DESCRIPTION
**Pull request by SniperSister**
Pull Request for Issue #37413 

### Summary of Changes
Remove the auth binding field and checking code to disable the feature until a proper solution has been implemented see #37405.

### Testing Instructions
See #34713

### Actual result BEFORE applying this Pull Request
You are NOT logged into the site

### Expected result AFTER applying this Pull Request
You are logged into the site

### CODE REVIEW

**Comment by brianteeman** [comment]
Tested the plugin changes and they at least let me log back in again. Have not tested the script.php changes which I assume is to fix upgrades. Still doesn't address the myriad of other issues with this disaster of a code change

**Comment by zero-24** [comment]
I have tested this item :white_check_mark: successfully on deb4d5f5ebcd701ae39c14cf4eae865b8e347f2e<br /><br />Confirmed the issue, tested the patch. Confirmed that the script.php checks the values and resets them back to empty when they are invalid<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37415">issues.joomla.org/tracker/joomla-cms/37415</a>.</sub>

File: administrator/components/com_admin/script.php [hunk 1]
```diff
@@ -8685,4 +8688,37 @@ protected function addUserAuthProviderColumn(): void
 			return;
 		}
 	}
+
+	/**
+	 * Verifies the current authProvider values in the user table and unsets invalid values
+	 *
+	 * @throws Exception
```
Comment (bembelimen) [inline administrator/components/com_admin/script.php]:
```suggestion
	 * @throws Exception
	 *
	 * @return  void
	 *
	 * @since  __DEPLOY_VERSION__
```

**Comment by bayareajenn** [comment]
I have tested this item :white_check_mark: successfully on deb4d5f5ebcd701ae39c14cf4eae865b8e347f2e<br /><br />I don't have a CA SSL cert so could not test Webauthn. I tested it the same way as 37416 and was able to duplicate the problem and then load the patch and login again.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37415">issues.joomla.org/tracker/joomla-cms/37415</a>.</sub>

**Comment by brianteeman** [comment]
Canyou remove the new language strings as well please
COM_USERS_USER_FIELD_AUTHPROVIDER_DESC
COM_USERS_USER_FIELD_AUTHPROVIDER_LABEL

File: plugins/user/joomla/joomla.php [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
 defined('_JEXEC') or die;
 
+use Joomla\CMS\Authentication\ProviderAwareAuthenticationPluginInterface;
```
Comment (brianteeman) [inline plugins/user/joomla/joomla.php]: ??

Comment (SniperSister) [reply plugins/user/joomla/joomla.php]: Fixed!

**Comment by SniperSister** [comment]
> Canyou remove the new language strings as well please

Done! Also checked in with @tecpromotion as tt-coordinator.

**Comment by brianteeman** [comment]
and remove
administrator/components/com_users/src/Field/PrimaryauthprovidersField.php

**Comment by brianteeman** [comment]
components/com_users/tmpl/profile/edit.php
`		<?php if (count($this->twofactormethods) > 1 && !empty($this->twofactorform)) : ?>`

not sure if that change  is still needed

**Comment by brianteeman** [comment]
`libraries/src/Authentication/ProviderAwareAuthenticationPluginInterface.php`

**Comment by brianteeman** [comment]
`libraries/src/Table/User.php`

	 * Updates auth provider of a user

**Comment by SniperSister** [comment]
>  administrator/components/com_users/src/Field/PrimaryauthprovidersField.php
> libraries/src/Authentication/ProviderAwareAuthenticationPluginInterface.php

Would leave those two files for now as removing them (and therefore adding them to the deletion list in script.php) would prevent us from re-using the same file names for the future patch.

File: administrator/components/com_users/src/Model/UserModel.php [hunk 3]
```diff
@@ -1217,12 +1217,6 @@ public function getTwofactorform($userId = null)
 		$userId = (!empty($userId)) ? $userId : (int) $this->getState('user.id');
 		$user = $this->getItem($userId);
```
Comment (Quy) [inline administrator/components/com_users/src/Model/UserModel.php]: Delete.

**Comment by brianteeman** [comment]
No you have to remove them (and come up with new names in future) otherwise you will be breaking semver when you change the contents of the library files.

**Comment by brianteeman** [comment]
libraries/src/User/User.php
public function setAuthProvider($authProvider)

**Comment by brianteeman** [comment]
plugins/authentication/cookie/cookie.php
use Joomla\CMS\Authentication\ProviderAwareAuthenticationPluginInterface;
public static function isPrimaryProvider()

and the same in other plugins

File: administrator/components/com_users/src/Model/UserModel.php (hunk 3 above)
Comment (SniperSister) [reply administrator/components/com_users/src/Model/UserModel.php]: Done!

**Comment by SniperSister** [comment]
> No you have to remove them (and come up with new names in future) otherwise you will be breaking semver when you change the contents of the library files.

Ok, fair comment, done!

**Comment by brianteeman** [comment]
Thanks @SniperSister

File: administrator/components/com_admin/script.php [hunk 4]
```diff
@@ -639,6 +639,9 @@ public function deleteUnexistingFiles($dryRun = false, $suppressOutput = false)
 		];
 
 		$files = array(
+			// From 4.1.1 to 4.1.2
+			'/administrator/components/com_users/src/Field/PrimaryauthprovidersField.php',
+
```
Comment (richard67) [inline administrator/components/com_admin/script.php]: Please move this to the end of the list.

**Comment by HLeithner** [comment]
> semver!!

where do you see a semver violation?

**Comment by brianteeman** [comment]
> > semver!!
> 
> where do you see a semver violation?

it should be properly marked as deprecated so that an ide etc will spot it.

**Comment by HLeithner** [comment]
> hat an ide etc will spot it.

The interface is deprecated all other parts get removed because not covered by our semver promise. Also semver doesn't allow us to add deprecations in bugfix releases. that's the reason David added the comment "Please note: might be deprecated with Joomla 4.2"

or do you mean something different?

**Comment by bembelimen** [comment]
Thanks everyone for this good collaboration.
@nikosdion and @brianteeman are you both ok with the solution?

**Comment by nikosdion** [comment]
@bembelimen I am OK with this solution.

**Comment by brianteeman** [comment]
@bembelimen seems ok.

**Comment by bembelimen** [comment]
Thank you all!

**Comment by dgrammatiko** [comment]
@SniperSister @bembelimen the binding issue has extreme similarities to the Media Manager Adapters, maybe you should take a look how @joomdonation @Fedik @laoneo and me tackled that binding and get some inspiration...

Basically https://github.com/joomla/joomla-cms/pull/34069 and https://github.com/joomla/joomla-cms/pull/33724 (maybe a couple more)

**Pull request merged**